src/communication/tlsvchannel.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "mbedtls/debug.h"
+#include <psa/crypto.h>
+
+#include "aos/common/crypto/mbedtls/driverwrapper.hpp"
+
+#include "tlsvchannel.hpp"
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+TLSVChannel::TLSVChannel(aos::SharedPtr<aos::crypto::x509::CertificateChain> certChain,
+    const aos::Array<uint8_t>& caCert, const aos::crypto::PrivateKeyItf& privKey, FuncDbg dbg)
+    : mCertChainPtr(certChain)
+    , mCACertArray(caCert)
+    , mPrivKey(privKey)
+    , mKeyID(0)
+    , mDbg(dbg)
+{
+}
+
+TLSVChannel::~TLSVChannel()
+{
+    mbedtls_x509_crt_free(&mCertChain);
+    mbedtls_x509_crt_free(&mCACert);
+    mbedtls_pk_free(&mPrivKeyCtx);
+    mbedtls_entropy_free(&mEntropy);
+    mbedtls_ctr_drbg_free(&mCtrDrbg);
+    mbedtls_ssl_config_free(&mConf);
+    mbedtls_ssl_free(&mSsl);
+
+    AosPsaRemoveKey(mKeyID);
+}
+
+aos::Error TLSVChannel::Init(const aos::String& xsReadPath, const aos::String xsWritePath)
+{
+    psa_status_t status = psa_crypto_init();
```

Comment:
It is called when we initialize crypto provider. Do we need to initialize psa again?

Reply:
No, but then we need to follow initialization order, so first crypto provider and then TLSVChannel